dbt_subprojects/solana/models/_sector/dex/phantom_swapper/phantom_swapper_solana_fee_payments_raw.sql

```diff
@@ -29,21 +28,23 @@ with sol_payments as (
                 fee_addresses.fee_receiver = account_activity.address
                 and balance_change > 0
             )
-        join {{ source('dex_solana', 'trades') }} as trades
-            on trades.tx_id = account_activity.tx_id
-            and trades.block_time = account_activity.block_time
-            {% if is_incremental() %} 
-            and {{ incremental_predicate('trades.block_time') }}
-            {% else %} 
-            and trades.block_time >= timestamp '{{query_start_date}}'
-            {% endif %} 
         where
             {% if is_incremental() %} 
                 {{ incremental_predicate('account_activity.block_time') }}
             {% else %} 
                 account_activity.block_time >= timestamp '{{query_start_date}}'
             {% endif %} 
             and tx_success
+            and exists (
+                select 1 
+                from {{ source('dex_solana', 'trades') }} as trades
+                where trades.tx_id = account_activity.tx_id
```

Comment:
minor, but in the code removed you also had join condition for `block_time`. should we continue to add here and one below? think it may help with telling trino engine which partitions to search to expedite the join.

Reply:
Ive added this now thanks